staging/src/k8s.io/client-go/tools/record/event_test.go

```diff
@@ -561,8 +621,9 @@ func TestLotsOfEvents(t *testing.T) {
 		},
 	}
 
-	eventBroadcaster := NewBroadcasterForTests(0)
```

Comment:
This seems to be the only place where we were using `NewBroadcasterForTests`, so it might be worth removing it from the library in a future PR

Reply:
It is used externally, for example here: https://github.com/kubewharf/kubeadmiral/blob/ad23985b98ea6a8ee133bc4a2f3a3e587ee8a259/pkg/util/eventsink/eventsink_test.go#L99

Reply:
Is it worth keeping around though if we don't use it in k/k anymore? Alternative still exist for them with the NewBroadcaster method

Reply:
I don't want to break existing clients if we can avoid it. Keeping this function public is not a big burden.

Reply:
If we want to remove that function, it should be in a separate PR, and perhaps with some deprecation period.